test/widgets/login_test.dart

```diff
@@ -65,7 +67,140 @@ void main() {
     expectErrorFromText('email@example.com', ServerUrlValidationError.noUseEmail);
   });
 
-  // TODO test AddAccountPage
+  group('AddAccountPage', () {
+    late FakeApiConnection connection;
+    List<Route<dynamic>> pushedRoutes = [];
+    List<Route<dynamic>> poppedRoutes = [];
+
+    List<Route<dynamic>> takePushedRoutes() {
+      final routes = pushedRoutes.toList();
+      pushedRoutes.clear();
+      return routes;
+    }
+
+    Future<void> prepare(WidgetTester tester) async {
+      addTearDown(testBinding.reset);
+
+      pushedRoutes = [];
+      poppedRoutes = [];
+      final testNavObserver = TestNavigatorObserver();
+      testNavObserver.onPushed = (route, prevRoute) => pushedRoutes.add(route);
+      testNavObserver.onPopped = (route, prevRoute) => poppedRoutes.add(route);
+      testNavObserver.onReplaced = (route, prevRoute) {
+        poppedRoutes.add(prevRoute!);
+        pushedRoutes.add(route!);
+      };
+
+      await tester.pumpWidget(ZulipApp(navigatorObservers: [testNavObserver]));
+      await tester.pump();
+      check(takePushedRoutes()).single.isA<WidgetRoute>().page.isA<ChooseAccountPage>();
+      await tester.tap(find.text('Add an account'));
+      check(takePushedRoutes()).single.isA<WidgetRoute>().page.isA<AddAccountPage>();
+      await testNavObserver.pumpPastTransition(tester);
+    }
+
+    testWidgets('happy path', (tester) async {
+      await prepare(tester);
+
+      final serverSettings = eg.serverSettings();
+
+      await tester.enterText(find.byType(TextField),
+        serverSettings.realmUrl.toString());
+      testBinding.globalStore.useCachedApiConnections = true;
+      connection = testBinding.globalStore.apiConnection(
+        realmUrl: serverSettings.realmUrl,
+        zulipFeatureLevel: null);
+      connection.prepare(json: serverSettings.toJson());
+      await tester.tap(find.text('Continue'));
+      await tester.pump(Duration.zero);
+      checkNoDialog(tester);
+      check(takePushedRoutes()).single.isA<WidgetRoute>().page.isA<LoginPage>()
+        .serverSettings.realmUrl.equals(serverSettings.realmUrl);
+    });
+
+    testWidgets('Server too old, well-formed response', (tester) async {
+      await prepare(tester);
+
+      final serverSettings = eg.serverSettings(
+        zulipFeatureLevel: 1, zulipVersion: '3.0');
+
+      testBinding.globalStore.useCachedApiConnections = true;
+      connection = testBinding.globalStore.apiConnection(
+        realmUrl: serverSettings.realmUrl,
+        zulipFeatureLevel: null);
+
+      await tester.enterText(find.byType(TextField),
+        serverSettings.realmUrl.toString());
+
+      connection.prepare(json: serverSettings.toJson());
```

Comment:
bump https://github.com/zulip/zulip-flutter/pull/1783#discussion_r2258489812 :slightly_smiling_face: 

Also I suspect this group of tests would be clarified by a little helper. Something like:
```dart
    await attempt(tester, serverSettings.realmUrl, serverSettings.toJson());
```

which would cover all these lines and the tap/pump below, and leave the caller to finish up with just the check steps.